### PR TITLE
quiz lquiz exam creationg mngt and grading

### PR DESCRIPTION
**Pull request by pdyba**

### CODE REVIEW

File: bootstrap.py [hunk 1]
```diff
@@ -321,9 +331,15 @@ def __str__(self):
         e_path = path + '.exercises'
         m_path = path + '.meta'
         q_path = path + '.quiz'
-        try:
+        try:  # lesson generation will BE detracted
```
[the remark this answers is not in the document]
Comment (pdyba) [reply bootstrap.py]: it is needed

File: frontend/src/components/_base/header.vue [hunk 2]
```diff
@@ -6,33 +6,39 @@
                     <!--<help v-if="auth"></help>-->
                     <navbar-item href="/about">About</navbar-item>
                     <navbar-item href="/lessons" v-if="auth">Lessons</navbar-item>
-                    <navbar-item href="/quiz" v-if="admin">Quiz</navbar-item>
-                    <navbar-item href="/live_quiz" v-if="admin">Live Quiz</navbar-item>
-                    <navbar-item href="/exam" v-if="admin">Exam</navbar-item>
+                    <!--<navbar-item href="/homework" v-if="auth">HomeWork</navbar-item>-->
```
[the remark this answers is not in the document]
Comment (pdyba) [reply frontend/src/components/_base/header.vue]: it is needed

File: frontend/src/components/_base/header.vue [hunk 3]
```diff
@@ -6,33 +6,39 @@
...
                         <dropdown-toggle tag="a" navLink color="indigo">Mentor</dropdown-toggle>
                         <dropdown-menu>
-                            <dropdown-item>Lesson</dropdown-item>
-                            <!--<dropdown-item>Quiz</dropdown-item>-->
-                            <!--<dropdown-item>Live Quiz</dropdown-item>-->
-                            <!--<dropdown-item>Users List</dropdown-item>-->
+                            <!--<dropdown-item href="/mentor/questions">Questions</dropdown-item>-->
+                            <!--<dropdown-item href="/mentor/lesson">Lesson: status</dropdown-item>-->
+                            <!--<dropdown-item href="/mentor/homework">Homework: grade</dropdown-item>-->
+                            <!--<dropdown-item href="/mentor/live_quiz">Live Quiz: Overview</dropdown-item>-->
```
[the remark this answers is not in the document]
Comment (pdyba) [reply frontend/src/components/_base/header.vue]: it is needed

File: frontend/src/components/_base/header.vue [hunk 4]
```diff
@@ -6,33 +6,39 @@
...
                     <dropdown tag="li" class="nav-item" v-if="org">
                         <dropdown-toggle tag="a" navLink color="indigo">Organiser</dropdown-toggle>
                         <dropdown-menu>
                             <!--<dropdown-item>Lesson: Create</dropdown-item>-->
-                            <dropdown-item href="/organiser/lessons">Lesson: List & Manage</dropdown-item>
-                            <!--<dropdown-item href="/organiser/attendee_review">Attendee: Review</dropdown-item>-->
-                            <!--<dropdown-item>Question: List & Manage</dropdown-item>-->
+                            <dropdown-item href="/organiser/attendee_review">Attendee: Review</dropdown-item>
+                            <dropdown-item href="/organiser/lessons">Lesson: Manage</dropdown-item>
+                            <!--<dropdown-item>Question: Manage</dropdown-item>-->
```
[the remark this answers is not in the document]
Comment (pdyba) [reply frontend/src/components/_base/header.vue]: it is needed

File: bootstrap.py [hunk 5]
```diff
@@ -449,11 +468,15 @@ def get_parser():
         loop.run_until_complete(gen_users())
     if args.bootstrap:
         loop.run_until_complete(bootstrap_db())
-        loop.run_until_complete(bootstrap_db())
+        ans = input('Have You seen any errors (red) - y/n ')
+        if ans.lower() == 'y':
+            loop.run_until_complete(bootstrap_db())
```
Comment (grzegorzpro) [inline bootstrap.py]: :D

File: models/common.py [hunk 6]
```diff
@@ -137,6 +137,11 @@ class Users(Table):
             self.magic_string_date = datetime.utcnow()
         return await super().update(**kwargs)
 
+    def is_only_attendee(self):
+        if self.admin or self.mentor or self.organiser:
+            return False
+        return True
```
Comment (grzegorzpro) [inline models/common.py]: return not (self.admin or self.mentor or self.organiser)

File: models/utils.py [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+            resp.append(quest)
+        return resp
+
+    async def delete_old_questions(self):
+        old_questions = self._questions(**{self._name: 1, 'question': 1, 'question_order': 1})
+        await old_questions.delete(**{self._name: self.id})
+
+    @classmethod
+    async def get_by_anwers_to_grade_by_qid(cls, qid):
+        anwares = []
```
Comment (grzegorzpro) [inline models/utils.py]: anwares?

File: orm.py [hunk 8]
```diff
@@ -381,21 +389,25 @@ def _format_kwargs(cls, **kwargs):
...
-                WHERE id={}
-                """.format(cls._name, data.id))
+            if hasattr(data, 'id'):
+                resp = await make_a_querry(
+                    """DELETE FROM {}
+                    WHERE id={}
+                    """.format(cls._name, data.id))
+            else:
+                resp = await make_a_querry("""DELETE FROM {} WHERE {}""".format(cls._name, cls._format_kwargs(**kwargs)))
+                print(resp)
```
Comment (grzegorzpro) [inline orm.py]: print

File: bootstrap.py (hunk 5 above)
Comment (pdyba) [reply bootstrap.py]: ;D

**Pull request merged**